### ISSUE TITLE
"'object' === typeof foo" Faliing

### ISSUE THREAD

**Issue by eksperimental** [eslint/eslint#3906]
the following code
`if ('object' === typeof event) {`

is giving me this error
> Expected literal to be on the right side of ===.

eventhough it is declared as a valid use in test:
https://github.com/eslint/eslint/blob/master/tests/lib/rules/valid-typeof.js#L29

**Comment by eslintbot** [comment]
Thanks for the issue! If you're reporting a bug, please be sure to include:

1. The version of ESLint you are using (run `eslint -v`)
2. What you did (the source code and ESLint configuration)
3. The actual ESLint output complete with numbers
4. What you expected to happen instead

Requesting a new rule? Please see [Proposing a New Rule](http://eslint.org/docs/developer-guide/contributing/new-rules) for instructions.

**Comment by BYK** [comment]
This looks like a `yoda` error to me.

**Comment by eksperimental** [comment]
Yes, thats what i meant

**Comment by lo1tuma** [comment]
That’s the expected behavior of the `yoda` rule.

**Comment by ilyavolodin** [comment]
As @lo1tuma noted, this is expected behavior. Try switching your code to:
```js
if (typeof event === 'object') {
```
